dags/synapse-dataset-to-croissant.py

```diff
@@ -292,272 +867,277 @@ def combine_dataset_lists(root_carrier_context: Dict, dataset_ids: List[List[Dic
-                        'datePublished', COALESCE(CREATED_ON, 'unknown_date'),
-                        'license', COALESCE(ANNOTATIONS:annotations:license:value[0], 'unknown_license'),
-                        'version', COALESCE(VERSION_NUMBER, 'unknown_version'),
-                        'distribution', ARRAY_PREPEND(ARRAY_AGG(distribution_pointing_to_files.file_object),
-                                            OBJECT_CONSTRUCT(
-                                                '@type', 'FileObject',
-                                                '@id', 'metadata',
-                                                'name', 'metadata',
-                                                'description', CONCAT(
-                                                    'Metadata associated with ', entity_metadata.DATASET_SYNAPSE_ID),
-                                                'contentUrl', CONCAT(
-                                                    'https://www.synapse.org/Synapse:', entity_metadata.DATASET_SYNAPSE_ID),
-                                                'encodingFormat', 'application/csv',
-                                                'md5', '',
-                                                'sha256', 'unknown'
-                                            )
-                                        ),
-                        'recordSet', fields_pointing_to_metadata.record_set
-                    ) AS metadata
-                FROM
-                    entity_metadata, fields_pointing_to_metadata
-                JOIN
-                    distribution_pointing_to_files
-                ON
-                    distribution_pointing_to_files.DATASET_SYNAPSE_ID = fields_pointing_to_metadata.DATASET_SYNAPSE_ID
-                GROUP BY
-                    entity_metadata.DATASET_SYNAPSE_ID,
-                    entity_metadata.name,
-                    fields_pointing_to_metadata.DATASET_SYNAPSE_ID,
-                    fields_pointing_to_metadata.record_set,
-                    entity_metadata.name,
-                    entity_metadata.ANNOTATIONS,
-                    entity_metadata.CREATED_ON,
-                    entity_metadata.VERSION_NUMBER
-            )
-
-            SELECT
-                metadata::STRING AS croissant_metadata,
-                DATASET_SYNAPSE_ID,
-                name
-            FROM
-                croissant_metadata;
-            """
-            snow_hook = SnowflakeHook(context["params"]["snowflake_developer_service_conn"])
-            ctx = snow_hook.get_conn()
-            cs = ctx.cursor()
-
-            otel_logger.info(
-                f"Performing query for dataset {dataset_id}")
+            span.set_attribute("airflow.dataset_id", f"{dataset_id}")
+            path_to_remove = None
+            syn_client = create_syn_client()
+            file_ids_and_versions_attached_to_dataset = []
 
             try:
```

Comment:
starting from this try and except block, could you add more comments in the code?

Reply:
I added a few more! [Take a look at this commit](https://github.com/Sage-Bionetworks-Workflows/orca-recipes/pull/83/commits/616a6e1d8cb81aa69762f59cffc6d452a0591f91)

Reply:
Great. Thanks for adding them.